lupa/tests/test.py

```diff
@@ -2620,6 +2620,19 @@ def test_nil_debug_traceback(self):
             self.assertNotIn("stack traceback:", e.args[0])
 
 
+################################################################################
+# tests for table access error
+
+class TestTableAccessError(SetupLuaRuntimeMixin, unittest.TestCase):
+    def test_error_index_metamethod(self):
+        self.lua.execute('''
+        t = {}
+        setmetatable(t, {__index = function() error() end})
+        ''')
+        lua_t = self.lua.eval('t')
+        self.assertRaises(lupa.LuaError, lambda t, k: t[k], lua_t, 'k')
```

Comment:
Can we somehow make sure that the `__index` method was really called, and that the error we receive is the one that we intentionally raised and nothing else? Otherwise, it's too easy to break something somewhere and still receive some kind of `LuaError` here, but a different one.

Reply:
Sure!

Reply:
Done in 58d2883